### PR TITLE
Vault backend

### PR DESCRIPTION
**Pull request by mourgaya**
create VAULT_BACKEND variable

vault_backend: [-consul-]{+"{{ lookup('env','VAULT_BACKEND') | default('consul', true) }}"+}
vault_backend_consul: vault_backend_consul.j2
vault_backend_file: vault_backend_file.j2
vault_backend_etcd: vault_backend_etcd.j2

### CODE REVIEW

**Comment by bbaassssiiee** [comment]
Purpose: allow to specify vault port and consul port and addr in case of multiple consul clusters in the same VLAN.

File: defaults/main.yml [hunk 1]
```diff
@@ -86,7 +86,9 @@ vault_backend_tls_key_file: "{{ vault_tls_key_file }}"
 vault_backend_tls_ca_file: "{{ vault_tls_ca_file }}"
 
 # Consul storage settings
-vault_consul: 127.0.0.1:8500
+consul_port:  "{{ lookup('env','CONSUL_PORT') | default('8500', true) }}"
+consul_addr: "{{ lookup('env','CONSUL_ADDR') | default('127.0.0.1', true) }}"
+vault_consul: "{{consult_addr}}:{{consul_port}}"
```
Comment (bbaassssiiee) [inline defaults/main.yml:91]: `consult_addr` is a typo, should be `consul_addr`.

**Review by bbaassssiiee — CHANGES_REQUESTED** [review]
Please fix the typo in `consult_addr`.

File: defaults/main.yml [hunk 2]
```diff
@@ -86,7 +86,9 @@ vault_backend_tls_key_file: "{{ vault_tls_key_file }}"
 vault_backend_tls_ca_file: "{{ vault_tls_ca_file }}"
 
 # Consul storage settings
-vault_consul: 127.0.0.1:8500
+consul_port:  "{{ lookup('env','CONSUL_PORT') | default('8500', true) }}"
+consul_addr: "{{ lookup('env','CONSUL_ADDR') | default('127.0.0.1', true) }}"
```
[the remark this answers is not in the document]
Comment (rgarrigue) [reply defaults/main.yml:90]: Vault should be running on other servers than Consul.

**Comment by rgarrigue** [comment]
Hi there. I'm coming around looking for something else : an option to enable disable_mlock. Since it's recommended for integrated storage (see https://www.vaultproject.io/docs/configuration#disable_mlock), maybe you can add it here ?

File: defaults/main.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (bbaassssiiee) [reply defaults/main.yml:90]: ... could be ... (with this PR)

**Comment by bbaassssiiee** [comment]
Closing due to neglect by contributor.

**Pull request closed without merging**